dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/QaServiceTest.kt

```diff
@@ -0,0 +1,47 @@
+package org.dataland.e2etests.tests
+
+import org.awaitility.Awaitility.await
+import org.dataland.datalandbackend.openApiClient.model.CompanyAssociatedDataEuTaxonomyDataForNonFinancials
+import org.dataland.datalandbackend.openApiClient.model.QAStatus as BackendQaStatus
+import org.dataland.datalandqaservice.openApiClient.model.QAStatus as QaServiceQaStatus
+import org.dataland.e2etests.auth.TechnicalUser
+import org.dataland.e2etests.utils.ApiAccessor
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import java.util.concurrent.TimeUnit
+
+class QaServiceTest {
+    private val apiAccessor = ApiAccessor()
+
+    private val testDataEuTaxonomyNonFinancials = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+        .getTData(1).first()
+
+    private val testCompanyInformation = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+        .getCompanyInformationWithoutIdentifiers(1).first()
+
+    @Test
+    fun `post dummy data, qa it and check the qa status changes`() {
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val storedCompanyInfos = apiAccessor.companyDataControllerApi.postCompany(testCompanyInformation)
+        val companyAssociatedEuTaxonomyNonFinancialsData =
+            CompanyAssociatedDataEuTaxonomyDataForNonFinancials(
+                storedCompanyInfos.companyId,
+                "",
+                testDataEuTaxonomyNonFinancials,
+            )
+
+        val dataId = apiAccessor.dataControllerApiForEuTaxonomyNonFinancials
+            .postCompanyAssociatedEuTaxonomyDataForNonFinancials(
+                companyAssociatedEuTaxonomyNonFinancialsData, false,
+            ).dataId
+
+        assertEquals(BackendQaStatus.pending, apiAccessor.metaDataControllerApi.getDataMetaInfo(dataId).qaStatus)
+
+        val qaServiceController = apiAccessor.qaServiceControllerApi
+        await().atMost(10, TimeUnit.SECONDS)
+            .until { qaServiceController.getUnreviewedDatasets().contains(dataId) }
+        qaServiceController.assignQualityStatus(dataId, QaServiceQaStatus.rejected)
+
+        assertEquals(BackendQaStatus.rejected, apiAccessor.metaDataControllerApi.getDataMetaInfo(dataId).qaStatus)
+    }
+}
```

Comment:
Should we also write a test where we let a dataset actually pass the QA?

Suggestion:
We could upload two datasets Alpha and Beta for one company in a `before` block.

Then test:  
- Assert that the `qaStatus` is "pending" if you get the meta info for both dataIds
- Assert that the dataIds are in the unreviewed-datasets-pool with the status set to "pending"
- Set dataset Alpha to "rejected", and assert that that the status changed in the meta info and in the unreviewed-datasets-pool
- Set dataset Beta to "accepted", and assert that that the status changed in the meta info and in the unreviewed-datasets-pool

To make the test short, you could write a method for the last two checks, since they are pretty similar.

Reply:
Currently all frontend e2e tests use the bypassQa. I added a backend e2e test that goes through the qa process and therefore does not use the bypassQa flag. However, I will have another look at the backend e2e tests and incorporate your suggestions. TO BE DONE

Reply:
I don't fully understand. Let's talk in the afternoon and clarify this.

Reply:
To be pushed